### PR TITLE
Move screen snippet to window

### LINKED ISSUE #275: Move ScreenSnippet functionality to WindowCore
From the GoogleDoc:
>ScreenSnippet: Perhaps this should just be a method on a window?

This currently takes a snippet of the _window_ not the screen. Given that this is all it is doing as well, this makes more sense to be a method on a window. It also has the side effect of lowering it's visibility, i.e. it will no longer be a top level concept.

### PR DESCRIPTION
**Pull request by JDawes-ScottLogic**
Remove `ScreenSnippet` class and implement on `WindowCore`.
Change implementations to capture the window referred to, rather than the calling window.

Resolves #275

### CODE REVIEW

File: packages/api-browser/src/window.ts [hunk 1]
```diff
@@ -214,5 +222,6 @@ const eventMap = {
   'focus': 'focus',
   'hide': 'hidden',
   'message': 'message',
-  'show': 'load'
+  'show': 'load',
+  'capture': 'capture'
```
Comment (andy-lee-eng) [inline packages/api-browser/src/window.ts]: I think we probably don't need this - these are mappings for window event names.

File: packages/api-demo/src/window/window-api-demo.js [hunk 2]
```diff
@@ -63,6 +63,10 @@ appReady.then(() => {
       win.addListener('close', () => {
         addListItem('close');
       });
+
+      win.addListener('capture', () => {
+        addListItem('capture');
```
Comment (andy-lee-eng) [inline packages/api-demo/src/window/window-api-demo.js]: Do we fire a Window event when capture happens?

Comment (JDawes-ScottLogic) [reply packages/api-demo/src/window/window-api-demo.js]: We don't currently. I'll add it to #301 .

File: packages/api-browser/src/window.ts (hunk 1 above)
Comment (JDawes-ScottLogic) [reply packages/api-browser/src/window.ts]: Agreed.

**Review by andy-lee-eng — APPROVED** [review]

**Pull request merged**